haskell/cabal.bzl

```diff
@@ -635,7 +646,11 @@ stack_snapshot = repository_rule(
     _stack_snapshot_impl,
     attrs = {
         "snapshot": attr.string(
-            doc = "The name of a Stackage snapshot.",
+            doc = "The name of a Stackage snapshot. Incompatible with local_snapshot.",
+        ),
+        "local_snapshot": attr.label(
+            doc = "A custom Stack snapshot file, as per the Stack documentation. Incompatible with snapshot.",
```

Comment:
This PR is a draft because I need to add example of this new attribute before merging. But it's already reviewable.